impl/src/main/java/com/okta/sdk/impl/resource/identity/provider/DefaultGoogleIdentityProviderBuilder.java

```diff
@@ -0,0 +1,144 @@
+        this.clientSecret = clientSecret;
+        return this;
+    }
+
+    @Override
+    public DefaultGoogleIdentityProviderBuilder setIsProfileMaster(Boolean isProfileMaster) {
+        this.isProfileMaster = isProfileMaster;
+        return this;
+    }
+
+    @Override
+    public DefaultGoogleIdentityProviderBuilder setMaxClockSkew(Integer maxClockSkew) {
+        this.maxClockSkew = maxClockSkew;
+        return this;
+    }
+
+    @Override
+    public DefaultGoogleIdentityProviderBuilder setUserNameTemplate(String userNameTemplate) {
+        this.userNameTemplate = userNameTemplate;
+        return this;
+    }
+
+    @Override
+    public DefaultGoogleIdentityProviderBuilder setPolicySubjectMatchType(PolicySubjectMatchType policySubjectMatchType) {
+        this.policySubjectMatchType = policySubjectMatchType;
+        return this;
+    }
+
+    @Override
+    public IdentityProvider buildAndCreate(Client client) {
+
+        IdentityProvider createdIdp = client.createIdentityProvider(client.instantiate(IdentityProvider.class)
+            .setType(IdentityProvider.TypeEnum.GOOGLE)
+            .setName(name)
+            .setProtocol(client.instantiate(Protocol.class)
+                .setType(protocolType)
+                .setScopes(scopes)
+                .setCredentials(client.instantiate(IdentityProviderCredentials.class)
+                    .setClient(client.instantiate(IdentityProviderCredentialsClient.class)
+                        .setClientId(clientId)
+                        .setClientSecret(clientSecret))))
+            .setPolicy(client.instantiate(IdentityProviderPolicy.class)
+                .setProvisioning(client.instantiate(Provisioning.class)
+                    .setAction(Provisioning.ActionEnum.AUTO)
+                    .setProfileMaster(isProfileMaster)
+                    .setGroups(client.instantiate(ProvisioningGroups.class)
+                        .setAction(ProvisioningGroups.ActionEnum.NONE))
+                    .setConditions(client.instantiate(ProvisioningConditions.class)
+                        .setDeprovisioned(client.instantiate(ProvisioningDeprovisionedCondition.class)
+                            .setAction(ProvisioningDeprovisionedCondition.ActionEnum.NONE))
+                        .setSuspended(client.instantiate(ProvisioningSuspendedCondition.class)
+                            .setAction(ProvisioningSuspendedCondition.ActionEnum.NONE))))
+                .setAccountLink(client.instantiate(PolicyAccountLink.class)
+                    .setFilter(null)
+                    .setAction(PolicyAccountLink.ActionEnum.AUTO))
+                .setSubject(client.instantiate(PolicySubject.class)
+                    .setUserNameTemplate(client.instantiate(PolicyUserNameTemplate.class)
+                        .setTemplate(userNameTemplate))
+                    .setMatchType(policySubjectMatchType))
+                .setMaxClockSkew(maxClockSkew)));
```

Comment:
wow, is this what the object actually looks like?

Reply:
Yeah, it isn't pretty :)